dorado/lagrangian_walker.py

```diff
@@ -38,11 +39,73 @@ def random_pick_seed(choices, probs=None):
     return choices[idx]
 
 
+def make_weight(Particles):
+    """Make an array with the routing weights."""
+    # init the weight array
+    L, W = np.shape(Particles.stage)
+    Particles.weight = np.zeros((L, W, 9))
+    # do weighting calculation for each cell
+    print('Calculating routing weights ...')
+    for i in tqdm(list(range(1, L-1)), ascii=True):
+        for j in list(range(1, W-1)):
+            # weights for each location in domain
+            # get stage values for neighboring cells
+            stage_ind = Particles.stage[i-1:i+2, j-1:j+2]
+
+            # calculate surface slope weights
+            weight_sfc = np.maximum(0,
```

Comment:
You should switch to an explicit import (`from numpy import maximum`) to shave off a bit more time

Reply:
Probably also `nansum`, maybe `zeros_like`, anything we're doing here a bunch of times

Reply:
I am not convinced that importing in this style is actually faster. I couldn't find anything online suggesting that it is the case? 

Ran some tests that basically ran code that looked like this (e.g. with `maximum()` function):
```
import time
import numpy as np
from numpy import maximum

t0 = time.time()
for i in range(500):
    np.maximum(arr, 1)
print('np time: ' + str(time.time() - t0))

t0 = time.time()
for i in range(500):
    maximum(arr, 1)
print('func time: ' + str(time.time() - t0))
```

And got results that were super close in times with the `np.maximum()` version sometimes fast and sometimes slower than the `maximum()` version.

Reply:
I learned this a few months ago when looking into optimizing something in `run_iteration`. People on stack said it can matter if you're looking up a function many thousands of times, which given the size of our domains seems feasible. Do you see a time difference if you change 500 to like 1e6? 
https://stackoverflow.com/questions/32151193/is-there-a-performance-cost-putting-python-imports-inside-functions

Reply:
It wasn't a big deal so I just imported the commonly called numpy functions within the function. 

For tests I did on 10,000 calls, the differences in time were <0.5s, and still flip-flopped a bit (maybe due to other CPU overhead at the time), but the change is simple and I don't think it makes the function any more or less confusing.